src/content/blog/2024-12-04-differences-between-linting-and-type-checking.md

```diff
@@ -0,0 +1,248 @@
+---
+layout: post
+title: "Differences between linting and type checking"
+teaser: "Linters such as ESLint and type checkers such as TypeScript catch different areas of code defects — and are best used in conjunction with each other."
+authors:
+  - joshuakgoldberg
+categories:
+  - Storytime
+tags:
+  - Linting
+---
+
+Linters and type checkers are two kinds of tools that both analyze code and report on detected issues.
+They may seem similar at first.
+However, the two kinds of tools detect very different issues and are useful in different ways.
+
+## Definitions
+
+*"Static analysis"* is the term for tooling that checks code for issues without running it.
```

Comment:
Stylistically, let's just use italics when introducing new terms (no quotes).

Also, static analysis isn't tooling, it's what the tooling does, so probably more accurately:

```suggestion
*Static analysis* is the inspection of source code without executing it. This differs from *dynamic analysis*, in which source code is inspected while it is executed. As such, dynamic analysis brings with it the inherent danger of executing malicious code or creating side effects while static analysis is safe to execute regardless of the source code.
```